examples/tensorrt_llm/common/base_engine.py

```diff
@@ -534,29 +551,44 @@ async def generate(self, request: TRTLLMWorkerRequest):
                 f"Worker inputs: {worker_inputs}, disaggregated params: {disaggregated_params}"
             )
 
-            sampling_params = get_sampling_params(request.sampling_params)
+            sampling_params = get_sampling_params(
+                request.sampling_options.dict(), self._default_sampling_params
+            )
+            max_tokens = request.stop_conditions.max_tokens
+            if max_tokens:
+                sampling_params.max_tokens = max_tokens
+
             async for response in self._llm_engine.generate_async(
                 inputs=worker_inputs,
                 sampling_params=sampling_params,
                 disaggregated_params=disaggregated_params,
-                streaming=False
-                if self._server_type == ServerType.CTX
-                else request.streaming,
+                streaming=self._server_type != ServerType.CTX,
             ):
-                # Convert the disaggregated params to OAI format so
-                # it can be sent over the network.
-                response.outputs[
-                    0
-                ].disaggregated_params = DisaggregatedTypeConverter.to_oai_disaggregated_params(
-                    response.outputs[0].disaggregated_params
-                )
+                if response.finished and self._server_type != ServerType.CTX:
+                    yield {"finish_reason": "stop", "token_ids": []}
+                    break
+
+                if not response.outputs:
+                    yield {"finish_reason": "error", "token_ids": []}
+                    break
 
-                yield TRTLLMWorkerResponse(
-                    request_id=request.id,
-                    prompt_token_ids=response.prompt_token_ids,
-                    outputs=[asdict(response.outputs[0])],
-                    finished=response.finished,
-                ).model_dump_json(exclude_unset=True)
+                output = response.outputs[0]
+                next_total_toks = len(output.token_ids)
+                out = {"token_ids": output.token_ids[num_output_tokens_so_far:]}
```

Comment:
In generate worker, the output will contain what has been generated by context worker?

Reply:
Actually, it always contains all generated tokens?

Reply:
Yes. All generated tokens are returned by the engine at every iteration. Hence, we have logic here to only send the newly generated tokens only.
The ingress would take care of either streaming the tokens or collecting them before sending for non-streaming.